### ISSUE TITLE
BUG: Can set breakpoints, but Safari doesn't stop

### ISSUE THREAD

**Issue by Azhrei** [quoid/userscripts#498]
First, thank you for creating Userscripts!  I was a fan of GM way back when and needed something like it again and found this. :)

I'm currently using Safari Version 16.4 (18615.1.26.110.1) on macOS Ventura 13.3.1(a).

My user script loads and works on the target web page, but breakpoints set using the dev tools > Source tab don't stop when the code is invoked.  I can place `console.log()` statements immediately above and below a block of code and both of them produce output, but breakpoints on those lines or other lines within the block do not stop.

I also notice that the breakpoint marker itself is the normal blue color that I typically get, but a light/dark grey depending on whether it's disabled or enabled.

I've looked through the Develop menu, in particular the Experimental Features submenu, but it's not clear if any of those need to be enabled or disabled for breakpoints to work.

Any help you can provide would be... very, uh... helpful. 😉

**Comment by Azhrei** [comment]
More information...

I'm using the latest release shown here, 4.4.2, via the App Store install process.

I tried disabling all extensions and the debugger works again; the breakpoint markers are back to being blue.

I reenabled all extensions and turned off the injection of my particular user script and the breakpoint marker is blue (and works).

So something the extension is doing is the problem...?  I'll try creating a very simple script...  No, that doesn't work either.  A script with only `alert("Just a test");` has a grey breakpoint marker and doesn't stop when injected.

I'll do some more web searching on this and report anything I find.  (But my search-fu seems weak right now.)

**Comment by Azhrei** [comment]
Alright, two more and I'll shut up.  (Well, unless I think of something else relevant, but I'm about done for this topic until next week. :))

The `debugger` keyword does not enter the debugger either.

I _can_ set breakpoints within files under the Extensions category, such as `script.js` (ad blocker) and `content.js` (Userscripts).

**Comment by quoid** [comment]
I am able to use `debugger` statements in my user scripts and it seems to function as expected; pauses execution of the JS.

Maybe paste a simplified version of what you tried that was failing?

```js
// ==UserScript==
// @name        All Sites Test
// @description This is your new file, start writing code
// @match       *://*/*
// @inject-into content
// ==/UserScript==

console.log(1);
debugger;
console.log('all sites')
```

**Comment by Azhrei** [comment]
Tank you for the quick response!

I've tested with a _very_ similar user script.  Mine looks like this:
```javascript
// ==UserScript==
// @name        breakpoint-test
// @description Test to see whether a very simple userscript breaks the debugger
// @match       *://*/*
// ==/UserScript==

alert("Just a test...");
debugger;
alert("Another alert...");
```

When I load the page https://canyouseeme.org/ I receive the first alert, then the second alert.  The debugger never gains control.

If I set a breakpoint (as shown in the image, below), it also doesn't stop.  Note that the breakpoint marker is grey, both in the source window and in the breakpoint list in the upper-left panel.  (The messages in the console re: Google analytics are not relevant; the results are the same for every page I've tried, including one on a local Apache server that was just `<?php phpinfo(); ?>`.)

I have selected Develop > Experimental Features > Restore Defaults to make sure nothing weird was going on (because I'd tried changing a few things), but resetting didn't help.

As another test, I tried changing the `@match` pattern to `http://*/*` and saved it.  Now the debugger works again.  How's that for narrowing it down!?  (Well, maybe...)

There must be an option or setting somewhere, but given that turning off extensions (or visiting a page without this particular extension enabled) seems to work, I don't know what to look at next...??

<img width="1282" alt="Screenshot 2023-05-31 at 10 40 34 PM" src="https://github.com/quoid/userscripts/assets/4120974/ada1729c-ca26-4158-b5c8-bbba7193c572">

**Comment by Azhrei** [comment]
Wow, false alarm, I guess...

I had tried quitting Safari and restarting it.  Didn't help.

I had tried rebooting.  Also didn't help.

In desperation, I tried rebooting again after shutting Safari down clean, and **NOW IT WORKS**.

(_Sigh_)

Sorry to waste your time with this.  I'm guessing some Experimental Feature required restarting Safari, and when I did the Reset To Defaults, I must not have restarted Safari afterwards (or something like that).

If/when it happens again, I'll file a bug report with Apple.  In the meantime, I'm closing this issue.  Thanks again for taking a look at it so quickly!
